File: Problem_Set_1/credit.c
#include <cs50.h>
#include <stdio.h>

bool verify(long x);
void decision(bool valid, long no);

int main(void)
{
    long number = get_long("Number: ");

    bool validity = verify(number);
    decision(validity, number);
}

bool verify(long x)
{
    long sum = 0;
    int i = 0;
    long num = x;

    do
    {
        if (i % 2 == 0)
        {
            sum += num % 10;
        }
        else
        {
            long n = (num % 10) * 2;
            if (n > 9)
            {
                n = n % 10 + n / 10;
            }
            sum += n;
        }
        i++;
        num /= 10;
    }
    while (num > 0);

    if (sum % 10 == 0)
    {
        return (true);
    }
    else
    {
        return (false);
    }
}

void decision(bool valid, long no)
{
    if (valid && (no / 10000000000000 == 34 || no / 10000000000000 == 37))
    {
        printf("AMEX\n");
    }
    else if (valid && (no / 100000000000000 == 51 || no / 100000000000000 == 52 ||
                       no / 100000000000000 == 53 || no / 100000000000000 == 54 ||
                       no / 100000000000000 == 55))
    {
        printf("MASTERCARD\n");
    }
    else if (valid && (no / 1000000000000 == 4 || no / 1000000000000000 == 4))
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}

File: Problem_Set_6/credit.py
from cs50 import get_int


def main():
    number = get_int("Number: ")

    validity = verify(number)
    decision(validity, number)


def verify(x):
    sum = 0
    i = 0
    num = x

    while True:
        if i % 2 == 0:
            sum += num % 10
        else:
            n = (num % 10) * 2
            if n > 9:
                n = n % 10 + int(n / 10)
            sum += n
        i += 1
        num = int(num / 10)
        if num <= 0:
            break

    if sum % 10 == 0:
        return True
    else:
        return False


def decision(valid, no):
    if valid and (int(no / 10000000000000) == 34 or int(no / 10000000000000) == 37):
        print("AMEX")
    elif valid and (int(no / 100000000000000) == 51 or int(no / 100000000000000) == 52 or
                    int(no / 100000000000000) == 53 or int(no / 100000000000000) == 54 or
                    int(no / 100000000000000) == 55):
        print("MASTERCARD")
    elif valid and (int(no / 1000000000000) == 4 or int(no / 1000000000000000) == 4):
        print("VISA")
    else:
        print("INVALID")


main()
